docs/lang/articles/deployment/tutorial.md

```diff
@@ -4,185 +4,215 @@ sidebar_position: 1
+Taichi makes it easy to write high-performance programs with efficient parallelism, but in many applications we cannot simply deploy the Python scripts. Taichi offers a runtime library (TiRT) with a C interface as well as its C++ wrapper, so your Taichi kernels can be launched in any native application. In this tutorial, we'll walk through the steps to deploy a Taichi program in a C++ application.
 
 ## Overview
 
 ![AOT E2E](../static/assets/aot_tutorial.png)
 
-In Python, when you call a function decorated with `@ti.kernel`, Taichi immediately compiles the kernel and sends it to the device for execution. This is called just-in-time (JIT) compilation. But generally speaking, we don't want to compile the kernels on a mobile phone, or to expose the source code to the users. For this Taichi introduced ahead-of-time (AOT) compilation so that you can compile kernels on a development machine, and launch them on user devices via TiRT.
+In Python, when you call a function decorated with `@ti.kernel`, Taichi immediately compiles the kernel and sends it to the device for execution. This is called just-in-time (JIT) compilation. However, in general, we don't want to compile the kernels on a mobile phone, or to expose the source code to the users. For this Taichi introduced ahead-of-time (AOT) compilation so that you can compile kernels on a development machine, and launch them on user devices via TiRT.
+
+In summary, running a Taichi program in C++ applications involves two steps:
 
 1. Compile Taichi kernels from Python and save the artifacts.
-2. Load AOT modules with TiRT and launch them in your applications.
+2. Load the AOT modules with TiRT and launch them in your applications.
 
 Although this tutorial only demonstrates integrating Taichi in a C++ application, the C interface allows you to integrate TiRT with many other programming languages including C/C++, Swift, Rust, C# (via P/Invoke) and Java (via JNI).
 
-### 1. Write kernels for AOT compilation
+## Quick-Start
 
-A Taichi kernel describes two aspects of a computer program: The computation itself, and the data it operates on. Because we don't know what kind of data will be fed into the kernel before execution, we have to clearly annotate the argument types for the AOT compiler.
+In this section, we will write a Taichi kernel for generating an image with color gradient and deploy it in a C++ application. The C++ program firstly allocates memory for the image data, and then the loaded Taichi kernel calculates the RGB value for each pixel. Finally, the image is output with a plain text ppm format. In this case, the Taichi kernel acts as a fragment shader and shades each pixel in parallel.
```

Comment:
The reader might not know what you mean by "color gradient". Try not to coin any term. You might want to use the Julia Set fractal we already used in the Taichi repo. It's somehow an iconic example of Taichi.

Also put the discussion about C++ program allocating memory to its dedicated section after the C++ code. Otherwise the reader would find it hard to grasp what exactly you want to convey.

The same for PPM. The output image format is not something so important that we have to mention it this early.

Reply:
I'm not so certain about the fractal thing though. It's up to you to replace the algo